### PR TITLE
doc: add code review guide.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zimulala @hanfei1991 @tiancaiamao PTAL

**Comment by ngaut** [comment]
PTAL @Wenting0905

File: code_review_guide.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+* Make sure you are familiar with the packages the PR modifies.
+
+* Make sure you have enough continuous time to review the PR, use 300 LOC per hour to estimate.
+
+* Make sure you can follow the updates of the PR in the next few work days.
+
+* Read the description of the PR, if it's not easy to understand, ask the coder to improve it.
+
+* For a bug fix PR, if there is no test case, ask the coder to add tests.
```
Comment (zhexuany) [inline code_review_guide.md]: maybe "for a bug fix RP, ask contributor  add test case to catch such bug necessary" better.

File: code_review_guide.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+* For a performance PR, if no benchmark result is provided, ask the coder to add a benchmark result.
+
+
+## Things to check during the review process
+
+* Am I able to understand the purpose of each unit test?
```
Comment (zhexuany) [inline code_review_guide.md]: Am I able to understand the purpose of each uint test added in RP?

File: code_review_guide.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+* Do unit tests actually test that the code is performing the intended functionality?
```
Comment (zhexuany) [inline code_review_guide.md]: Do unit tests actually test the code changes want to merge into master branch?

File: code_review_guide.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+* Do unit tests cover all the important code blocks and specially handled errors?
```
Comment (zhexuany) [inline code_review_guide.md]: Do unit tests cover all the important codes blocks within code changes in such PR?

File: code_review_guide.md (hunk 3 above)
Comment (coocood) [inline code_review_guide.md]: Every PR want to merge into master.

File: code_review_guide.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+* Could procedure tests be rewritten to table driven tests?
+
+* Is the code written following the style guide?
+
+* Is the same code duplicated more than twice?
```
Comment (zhexuany) [inline code_review_guide.md]: Does some code appear more than on place? Usually this can be refactor as a function.

File: code_review_guide.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+* Do comments exist and describe the intent of the code?
```
Comment (zhexuany) [inline code_review_guide.md]: Do comments appear right above function and in go programming style. If yes, do they describe the intention of code changes?

File: code_review_guide.md (hunk 1 above)
Comment (coocood) [inline code_review_guide.md]: This not only applies to the contributors but also project members.

File: code_review_guide.md (hunk 5 above)
Comment (coocood) [inline code_review_guide.md]: Each listed item is a single question, this would break the style.

File: code_review_guide.md (hunk 6 above)
Comment (coocood) [inline code_review_guide.md]: Ditto

File: code_review_guide.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
+# Code Review Guide
+
+## Things to do before you start reviewing the PR
+
+* Make sure you are familiar with the packages the PR modifies.
+
+* Make sure you have enough continuous time to review the PR, use 300 LOC per hour to estimate.
```
Comment (shenli) [inline code_review_guide.md]: LOC?

File: code_review_guide.md (hunk 2 above)
Comment (coocood) [inline code_review_guide.md]: I think in this context, unit test means unit test added in the PR, code means the changes in the PR.

File: code_review_guide.md (hunk 4 above)
Comment (coocood) [inline code_review_guide.md]: Ditto

File: code_review_guide.md (hunk 7 above)
Comment (coocood) [inline code_review_guide.md]: lines of code

**Comment by Wenting0905** [comment]
LGTM

**Pull request merged**